x-pack/platform/plugins/shared/alerting/server/routes/rule/apis/list_types/external/transforms/transform_rule_types_response/v1.ts

```diff
@@ -27,9 +27,9 @@ export const transformRuleTypesResponse = (
         ? { does_set_recovery_context: ruleType.doesSetRecoveryContext }
         : {}),
       enabled_in_license: ruleType.enabledInLicense,
-      ...(ruleType.fieldsForAAD ? { fieldsForAAD: ruleType.fieldsForAAD } : {}),
+      fieldsForAAD: [],
```

Comment:
transforming to `fieldsForAAD: []` and `has_fields_for_a_a_d: true` as `RegistryRuleType` don't have these fields anymore